app/frontend/src/lib/ui/components/filterGroup.test.js

```diff
@@ -311,4 +318,56 @@ describe('filterGroup', () => {
       expect(document.getElementsByClassName('govuk-accordion__section--expanded').length).toEqual(3);
     });
   });
+
+  describe('addUpdateOpenOrCloseEvent', () => {
+    test('displays close all when all elements are open', () => {
+      document.body.innerHTML = `
+      <form data-auto-submit="true">
+        <div><button id="close-all-groups">Open all</button></div>
+        <div class="govuk-accordion__section govuk-accordion__section--expanded filter-group__container">
+          <div class="header"><h3 class="heading"><button class="govuk-accordion__section-button"></button></h3></div>
+        </div>
+        <div class="govuk-accordion__section govuk-accordion__section--expanded filter-group__container">
+          <div class="header"><h3 class="heading"><button class="govuk-accordion__section-button"></button></h3></div>
+        </div>
+        <div class="govuk-accordion__section filter-group__container">
+          <div class="header"><h3 class="heading"><button id="closed-section-button" class="govuk-accordion__section-button"></button></h3></div>
+        </div>
+      </form>
+      `;
+
+      const buttonElement = document.getElementById('closed-section-button');
+      const openOrCloseAllSelector = 'close-all-groups';
+
+      addUpdateOpenOrCloseEvent(buttonElement, openOrCloseAllSelector);
+      buttonElement.dispatchEvent(new Event('click'));
+
+      expect(document.getElementById(openOrCloseAllSelector).innerText).toEqual(CLOSE_ALL_TEXT);
+    });
+
+    test('displays open all when all elements are closed', () => {
+      document.body.innerHTML = `
+      <form data-auto-submit="true">
```

Comment:
see above comment

Reply:
Also is the only difference between these chunks of mark up the `__expanded` and `Close all` text? making a common function to generate these bits of markup making use of template variables `${variable}` could make this neater, easier to read and less repetitive